lib/remote/apilistener.hpp

```diff
@@ -257,6 +273,9 @@ class ApiListener final : public ObjectImpl<ApiListener>
 	mutable std::mutex m_ActivePackageStagesLock;
 	std::map<String, String> m_ActivePackageStages;
 
+	/* update/delete object config sync */
+	static std::unique_ptr<ConfigSyncRule> m_ConfigSyncPolicy;
```

Comment:
I don't get the naming here either. Also, why a `unique_ptr`?

```suggestion
	/* ensures that at most one create/update/delete is being processed per object at each time */
	mutable ConfigSyncRule m_ObjectConfigChangeLock;
```

Reply:
Without a pointer, you cannot use a static instance of `ObjectNamesMutex` as a member of `ApiListener`.  Making it `mutable` is also useless, since all methods that access this variable are static.

Reply:
> Without a pointer, you cannot use a static instance of `ObjectNamesMutex` as a member of `ApiListener`.

That should be possible. What prevents you from doing so?

> Making it `mutable` is also useless, since all methods that access this variable are static.

They are static, but they are doing `ApiListener::Ptr listener = ApiListener::GetInstance();` anyways.